From cfcebcad0d2ef4dfc7a4d7155a431cd53556b452 Mon Sep 17 00:00:00 2001
From: SrujanaChalla <schalla@marvell.com>
Date: Thu, 4 Apr 2019 15:12:07 +0530
Subject: [PATCH 142/386] crypto: cpt - Fixed incorrect lock usage

Modified code to use spink_lock_irqsave() instead of spin_lock(),
as spin_lock() could not be used when the code executing in both
interrupt_context and process_context. Ref: OCTEONTX-2501

Change-Id: I6cef4bb67f2ed228d2eedc335676da0425851359
Signed-off-by: SrujanaChalla <schalla@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/6982
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/crypto/cavium/cpt/8x/cpt8x_reqmgr.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/cavium/cpt/8x/cpt8x_reqmgr.c b/drivers/crypto/cavium/cpt/8x/cpt8x_reqmgr.c
index 02781b7f46b7..37ec802746e3 100644
--- a/drivers/crypto/cavium/cpt/8x/cpt8x_reqmgr.c
+++ b/drivers/crypto/cavium/cpt/8x/cpt8x_reqmgr.c
@@ -103,10 +103,11 @@ static void cpt8x_send_cmd(union cpt_inst_s *cptinst, u32 db_count, void *obj)
 	struct cpt_vf *cptvf = (struct cpt_vf *) obj;
 	struct command_qinfo *qinfo = &cptvf->cqinfo;
 	struct command_queue *queue = &qinfo->queue[0];
+	unsigned long flags;
 	u8 *ent;
 
 	/* lock commad queue */
-	spin_lock(&queue->lock);
+	spin_lock_irqsave(&queue->lock, flags);
 	ent = &queue->qhead->head[queue->idx * qinfo->cmd_size];
 	memcpy(ent, (void *) cptinst, qinfo->cmd_size);
 
@@ -123,7 +124,7 @@ static void cpt8x_send_cmd(union cpt_inst_s *cptinst, u32 db_count, void *obj)
 	smp_wmb();
 	cptvf_write_vq_doorbell(cptvf, db_count);
 	/* unlock command queue */
-	spin_unlock(&queue->lock);
+	spin_unlock_irqrestore(&queue->lock, flags);
 }
 
 void cpt8x_send_cmds_in_batch(union cpt_inst_s *cptinst, u32 num, void *obj)
-- 
2.17.1

